mubench.pipeline/requirements.py

```diff
@@ -74,3 +77,56 @@ def __init__(self):
 
     def check(self):
         pass
+
+class CPUCountRequirement(Requirement):
+    MIN_CPU_COUNT = 2
+
+    def __init__(self):
+        super().__init__("CPUs >= {}".format(self.MIN_CPU_COUNT))
+
+    def check(self):
+        cpu_count = self._get_cpu_count()
+        if cpu_count < self.MIN_CPU_COUNT:
+            raise ValueError("Only {} CPUs available.".format(cpu_count))
+
+    def _get_cpu_count(self):
+        if _in_container():
+            return self._get_container_cpu_count()
+        else:
+            return psutil.cpu_count(logical=False)
```

Comment:
I'm not sure what you mean.
_What happens when we run this check inside a docker container?_ Docker is so nice as to mount these system files for each container. Hence, we just read the information we need from the respective files.
_Why can we not just use psutil?_ Container do not have their "own" system environment. They just have a limited scope of the host system they are allowed to access. When psutil accesses the systems specs, it still sees the specs of the host system and not the constraints of the container.

Hopefully I covered you're question with that. Does this solution also work on mac? I'm not sure if the mounted folder structure is the same there.